src/controllers/organizationController.ts

```diff
@@ -414,6 +378,63 @@ export default class OrganizationController {
     return none;
   }
 
+  private async createOnboardingDocuments(
+    ipaCode: string,
+    requestsCreationResponseSuccess: IResponseSuccessCreation<RequestCollection>
+  ): Promise<
```

Comment:
try to return a TaskEither here (you can wrap inner promises with `tryCatch`). and instead of call Promise.all, sequence an array of TaskEithers:

```ts
import { taskEither, TaskEither } from "fp-ts/lib/TaskEither";
import { array } from "fp-ts/lib/Array";

array.sequence(taskEither)(<array of taskeithers>) // returns Either<first error, latest value>
```

generally, try to avoid working with promises directly otherwise you have to wrap / unwrap eithers continously and/or deal with await / try / catch

see https://gcanti.github.io/fp-ts/recipes/async.html